### PR TITLE
Delay reg free for tbx_1reg

### LINKED ISSUE #86234: Test failure _Vector64_ro::JIT.HardwareIntrinsics.General._Vector64.Program.WidenInt64()
**Failed in:** runtime-coreclr jitstress2-jitstressregs 20230514.1

**Failed tests:**
```
coreclr osx arm64 Checked jitstress2_jitstressregs0x80 @ OSX.1200.ARM64.Open
    - _Vector64_ro::JIT.HardwareIntrinsics.General._Vector64.Program.WidenInt64()
    - _AdvSimd_Arm64_ro::JIT.HardwareIntrinsics.Arm._AdvSimd.Arm64.Program.VectorTableLookupExtension_Vector128_SByte()
    - _AdvSimd_ro::JIT.HardwareIntrinsics.Arm._AdvSimd.Program.VectorTableLookupExtension_Vector64_Byte()
    - _AdvSimd_ro::JIT.HardwareIntrinsics.Arm._AdvSimd.Program.VectorTableLookupExtension_Vector64_SByte()
coreclr windows arm64 Checked jitstress2_jitstressregs0x80 @ Windows.11.Arm64.Open
    - _Vector64_ro::JIT.HardwareIntrinsics.General._Vector64.Program.WidenInt64()
    - _AdvSimd_Arm64_ro::JIT.HardwareIntrinsics.Arm._AdvSimd.Arm64.Program.VectorTableLookupExtension_Vector128_SByte()
    - _AdvSimd_ro::JIT.HardwareIntrinsics.Arm._AdvSimd.Program.VectorTableLookupExtension_Vector64_Byte()
    - _AdvSimd_ro::JIT.HardwareIntrinsics.Arm._AdvSimd.Program.VectorTableLookupExtension_Vector64_SByte()
coreclr linux arm64 Checked jitstress2_jitstressregs0x80 @ (Ubuntu.1804.Arm64.Open)Ubuntu.1804.Armarch.Open@mcr.microsoft.com/dotnet-buildtools/prereqs:ubuntu-18.04-helix-arm64v8
    - _Vector64_ro::JIT.HardwareIntrinsics.General._Vector64.Program.WidenInt64()
    - _AdvSimd_Arm64_ro::JIT.HardwareIntrinsics.Arm._AdvSimd.Arm64.Program.VectorTableLookupExtension_Vector128_SByte()
    - _AdvSimd_ro::JIT.HardwareIntrinsics.Arm._AdvSimd.Program.VectorTableLookupExtension_Vector64_Byte()
    - _AdvSimd_ro::JIT.HardwareIntrinsics.Arm._AdvSimd.Program.VectorTableLookupExtension_Vector64_SByte()
```

**Error message:**
```
 One or more scenarios did not complete as expected.
```

**Stack trace:**
```
    at JIT.HardwareIntrinsics.General._Vector64.Program.WidenInt64() in /__w/1/s/artifacts/tests/coreclr/obj/AnyOS.x64.Checked/Managed/JIT/HardwareIntrinsics/General/Vector64/Vector64_ro/Vector64_ro/gen/Widen.Int64.cs:line 52
   at Program.<<Main>$>g__TestExecutor47|0_46(StreamWriter tempLogSw, StreamWriter statsCsvSw, <>c__DisplayClass0_0&)
```

### PR DESCRIPTION
**Pull request by kunalspathak**
- Setting the operand as "delay reg free" was missing for the scenario of `VectorTableLookupExtension()` that takes 1 row. Fixed that.
- Also we were incorrectly setting the return type of `WidenLower()` to `SIMD8` instead of `SIMD16`. Fixed that as well.

Fixes: #86234

### CODE REVIEW

File: src/coreclr/jit/lsraarm64.cpp [hunk 1]
```diff
@@ -1773,7 +1773,33 @@ int LinearScan::BuildConsecutiveRegistersForUse(GenTree* treeNode, GenTree* rmwN
...
     else
     {
-        srcCount += BuildOperandUses(treeNode);
+        RefPositionIterator refPositionMark   = refPositions.backPosition();
+        int                 refPositionsAdded = BuildOperandUses(treeNode);
+
+        if (rmwNode != nullptr)
+        {
+            // Check all the newly created Refpositions for delay free
+            RefPositionIterator tailRefPos = refPositions.backPosition();
```
Comment (BruceForstall) [inline src/coreclr/jit/lsraarm64.cpp:1782]: tailRefPos is unused?

Comment (kunalspathak) [reply src/coreclr/jit/lsraarm64.cpp]: fixed.

**Comment by kunalspathak** [comment]
@tannergooding for the `WidenLower` fix.

**Review by tannergooding — APPROVED** [review]

**Comment by kunalspathak** [comment]
@BruceForstall - do you want to take another look?

**Review by BruceForstall — APPROVED** [review]

**Pull request merged**